### PR TITLE
MS performance testing

### PR DESCRIPTION
**Pull request by CatEek**
## Description

> [!NOTE]  
> **tldr**: 

Debugging MS in Lightning API. 

This is an intermediate PR focusing on bug fixes and minor refactorings.

### Background - why do we need this PR?

Fixed stitching, normalization, np.concat -> concatenate changed for backward compatibility. In pyproject numpy is only forced to 2.x + if python is >= 3.13

## How has this been tested?

PSNR on HT_LIF24 is comparable to original values.

PSNR on Highres 28.61+-0.348	32.24+-0.647
MicroSSIM on Highres 0.804+-0.018	0.895+-0.010
MicroS3IM on Highres 0.905+-0.006	0.940+-0.007
SSIM on Highres 0.672+-0.012	0.708+-0.011
MSSSIM on Highres 0.830+-0.006	0.767+-0.013

## Additional Notes and Examples

Datamodule class is a mess. Needs a refactoring in NG dataset scope. And example data is also a mess, we need to discuss the setup. Messy train/val split + unclear matching with high_snr data
---

**Please ensure your PR meets the following requirements:**

- [x] Code builds and passes tests locally, including doctests
- [ ] New tests have been added (for bug fixes/features)
- [x] Pre-commit passes
- [ ] PR to the documentation exists (for bug fixes / features)

### CODE REVIEW

File: src/careamics/lightning/microsplit_data_module.py [hunk 1]
```diff
@@ -153,23 +152,24 @@ def get_train_val_data(
...
-    elif datasplit_type == DataSplitType.Val:
-        data = data[val_idx].astype(np.float64)
-    elif datasplit_type == DataSplitType.Test:
-        # TODO this is only used for prediction, and only because old dataset uses it
-        data = data[test_idx].astype(np.float64)
-    else:
-        raise Exception("invalid datasplit")
-
-    return data
+    # train_idx, val_idx, test_idx = get_datasplit_tuples(
```
Comment (veegalinova) [inline src/careamics/lightning/microsplit_data_module.py]: Is this commented out intentionally? If so, let's delete all of this?

File: src/careamics/lightning/microsplit_data_module.py [hunk 2]
```diff
@@ -523,6 +498,11 @@ def setup(self, stage: str | None = None) -> None:
         )
         self.predict_dataset.set_mean_std(*self.pred_config.data_stats)
 
+    # def prepare_data(self) -> None:
```
Comment (veegalinova) [inline src/careamics/lightning/microsplit_data_module.py]: Same, delete?

File: src/careamics/lvae_training/dataset/lc_dataset.py [hunk 3]
```diff
@@ -261,7 +261,7 @@ def __getitem__(self, index: Union[int, tuple[int, int]]):
 
         norm_target = self.normalize_target(target)
 
-        output = [inp, norm_target]
+        output = [inp, norm_target, index]
```
Comment (veegalinova) [inline src/careamics/lvae_training/dataset/lc_dataset.py]: This should not break anything, but I did not find where the index is used?

File: src/careamics/lvae_training/dataset/multich_dataset.py [hunk 4]
```diff
@@ -208,6 +208,9 @@ def load_data(
             test_fraction=test_fraction,
             allow_generation=allow_generation,
         )
+
+        # self._data = self._data[..., ::-1]
```
Comment (veegalinova) [inline src/careamics/lvae_training/dataset/multich_dataset.py]: Again, is it commented intentionally?

File: src/careamics/lvae_training/metrics.py [hunk 5]
```diff
@@ -1,214 +1,424 @@
...
 
 import numpy as np
 import torch
 from skimage.metrics import structural_similarity
+from torch import Tensor
 from torchmetrics.image import MultiScaleStructuralSimilarityIndexMeasure
 
 from careamics.models.lvae.utils import allow_numpy
 
+try:
```
Comment (veegalinova) [inline src/careamics/lvae_training/metrics.py]: looks a bit complicated, is it nessesary? would something like this work? 
```suggestion
try:
    from microssim import MicroMS3IM, MicroSSIM
except ImportError as exc:
    raise ImportError(...) from exc
```

File: src/careamics/lvae_training/metrics.py [hunk 6]
```diff
@@ -1,214 +1,424 @@
...
+    _microssim = importlib.import_module("microssim")
```
Comment (veegalinova) [inline src/careamics/lvae_training/metrics.py]: Should microSSIM be added to optional project dependencies?

**Review by veegalinova — CHANGES_REQUESTED** [review]

File: src/careamics/lightning/microsplit_data_module.py (hunk 1 above)
Comment (CatEek) [reply src/careamics/lightning/microsplit_data_module.py]: Removed

File: src/careamics/lvae_training/metrics.py (hunk 6 above)
Comment (CatEek) [reply src/careamics/lvae_training/metrics.py]: added

Comment (CatEek) [reply src/careamics/lvae_training/metrics.py]: #685

File: src/careamics/config/configuration_factories.py [hunk 7]
```diff
@@ -1839,28 +1840,44 @@ def create_hdn_configuration(
...
     output_channels: int = 1,
-    encoder_n_filters: int = 32,
-    decoder_n_filters: int = 32,
-    encoder_dropout: float = 0.0,
-    decoder_dropout: float = 0.0,
+    encoder_dropout: float = 0.1,
+    decoder_dropout: float = 0.1,
     nonlinearity: Literal[
         "None", "Sigmoid", "Softmax", "Tanh", "ReLU", "LeakyReLU", "ELU"
-    ] = "ReLU",  # TODO do we need all these?
```
Comment (jdeschamps) [inline src/careamics/config/configuration_factories.py:1860]: I may have written the TODO that disappeared. Do we really need to allow the choice of so many activation function? If all our examples and use cases use ELU, then we should just stick to that and not allow users to change it via the convenience function.

**Review by jdeschamps — APPROVED** [review]

**Comment by jdeschamps** [comment]
@veegalinova are you happy with the resolution of your comments?

**Comment by veegalinova** [comment]
> @veegalinova are you happy with the resolution of your comments?

The only thing that is left is a weird import in `metrics.py`, I reopened the comment

File: src/careamics/lvae_training/metrics.py (hunk 5 above)
Comment (veegalinova) [reply src/careamics/lvae_training/metrics.py]: now it's a dependency, so the chek is not needed at all?

Comment (CatEek) [reply src/careamics/lvae_training/metrics.py]: fixed

**Review by veegalinova — APPROVED** [review]
